src/Development/Rattle/Types.hs

```diff
@@ -52,11 +53,14 @@ fsaTrace t fs = nubTrace . mconcat $ map f fs
         f (FSATouch x) = g [] [x]
 
 nubTrace :: Ord a => Trace a -> Trace a
-nubTrace (Trace t a b) = Trace t (nubOrd a \\ b) (nubOrd b)
+nubTrace (Trace t r a b) = Trace t r (nubOrd a \\ b) (nubOrd b)
 
 
 newtype T = T Int -- timestamps
-    deriving (Enum,Eq,Ord,Show)
+    deriving (Enum,Eq,Ord,Show,Read)
```

Comment:
I think deriving Hashable should be OK, It works via GeneralizedNewtypeDeriving.

Reply:
Are you sure? 

>  No instance for (Generic T)
        arising from the 'deriving' clause of a data type declaration
      Possible fix:
        use a standalone 'deriving instance' declaration,
          so you can specify the instance context yourself
deriving (Enum,Eq,Ord,Show,Read,Hashable)